retrieval/scrape.go

```diff
@@ -833,12 +877,9 @@ func (sl *scrapeLoop) reportStale(start time.Time) error {
 }
 
 func (sl *scrapeLoop) addReportSample(app storage.Appender, s string, t int64, v float64) error {
-	re, ok := sl.refCache[s]
-
+	ref, ok := sl.cache.getRef(s)
```

Comment:
this is only unsafe really, if people expose the report metrics, which will probably cause funkyness regardless.